tests/app/celery/test_process_delivery_status_result_tasks.py

```diff
@@ -17,196 +26,358 @@ def sample_delivery_status_result_message():
+
+
+# confirmed: task should retry when incoming message has invalid provider attribute
+def test_celery_event_with_invalid_provider_attribute(notify_db_session, sample_delivery_status_result_message):
+    """Test that celery will retry the task if "message" is invalid from the CeleryEvent message"""
+
+    # remove provider (key) from the sample_delivery_status_result_message
+    sample_delivery_status_result_message.get("message")["provider"] = "abc123"
+
+    with pytest.raises(Retry):
+        process_delivery_status_result_tasks.process_delivery_status(event=sample_delivery_status_result_message)
+
+
+# confirm: task should retry when incoming message does not contain body attribute
+def test_celery_event_with_invalid_body_attribute(notify_db_session, sample_delivery_status_result_message):
+    """Test that celery will retry the task if "message" is missing from the CeleryEvent message"""
 
+    # remove body (key) from the sample_delivery_status_result_message
+    sample_delivery_status_result_message.get("message")["body"] = "body"
 
-def test_without_provider(
-    mocker,
-    notify_db_session,
-    sample_delivery_status_result_message,
-    sample_translate_return_value,
-    sample_notification,
+    with pytest.raises(Retry):
+        process_delivery_status_result_tasks.process_delivery_status(event=sample_delivery_status_result_message)
+
+
+# test get_provider_info when no provider is given by the celery event
+def test_get_provider_info_with_no_provider(
+        notify_db_session,
+        sample_notification_platform_status,
+        sample_sqs_message_without_provider
 ):
-    """Test that celery will retry the task if provider doesnt exist then self.retry is called"""
+    mock_celery_task = MockCeleryTask()
 
-    # change message['provider'] to invalid provider name
-    sample_delivery_status_result_message["message"]["provider"] = "abc"
-    mocker.patch("app.clients")
-    mocker.patch("app.clients.sms.SmsClient")
-    mocker.patch(
-        "app.clients.sms.twilio.TwilioSMSClient.translate_delivery_status",
-        return_value=sample_translate_return_value,
+    # now supply the sample to the function we want to test
+    with pytest.raises(Retry):
+        process_delivery_status_result_tasks._get_provider_info(mock_celery_task, sample_sqs_message_without_provider)
+
+
+def test_get_provider_info_with_invalid_provider(
+        notify_db_session,
+        sample_sqs_message
+):
+    """Test that _get_provider_info() will raise a celery retry when sqs message has an invalid provider"""
+
+    # mock the celery task event
+    mock_celery_task = MockCeleryTask()
+    sample_sqs_message['provider'] = "abc"
+
+    # now supply the sample to the function we want to test with the expectation of failure
+    with pytest.raises(Retry):
+        process_delivery_status_result_tasks._get_provider_info(mock_celery_task, sample_sqs_message)
+
+
+# here we test all valid provider names
```

Comment:
are you planning to add more provider names? there is no need to parameterize if you aren't.